src/coreclr/vm/prestub.cpp

```diff
@@ -503,21 +503,33 @@ PCODE MethodDesc::GetPrecompiledR2RCode(PrepareCodeConfig* pConfig)
 
     PCODE pCode = NULL;
 #ifdef FEATURE_READYTORUN
-    Module * pModule = GetModule();
+    Module * pModule = GetLoaderModule();
     if (pModule->IsReadyToRun())
     {
         pCode = pModule->GetReadyToRunInfo()->GetEntryPoint(this, pConfig, TRUE /* fFixups */);
     }
 
-    // Lookup in the entry point assembly for a R2R entrypoint (generics with large version bubble enabled)
-    if (pCode == NULL && HasClassOrMethodInstantiation() && SystemDomain::System()->DefaultDomain()->GetRootAssembly() != NULL)
+    //  Generics may be located in several places
+    if (pCode == NULL && HasClassOrMethodInstantiation())
     {
-        pModule = SystemDomain::System()->DefaultDomain()->GetRootAssembly()->GetModule();
-        _ASSERT(pModule != NULL);
+        Module* pDefiningModule = GetModule();
+        // Lookup in the defining module of the generic (which is where in inputbubble scenarios
```

Comment:
Wording nit - I think that the word "that" at line 517 is superfluous and there's a missing closing parenthesis.